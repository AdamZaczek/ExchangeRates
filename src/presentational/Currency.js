import React from 'react';
import Button from './Button';

const Currency = props => {
  const { currency, ask, bid, code } = props.currency;
  return (
    <tr>
      <td>{currency}</td>
      <td>{ask}</td>
      <td>{bid}</td>
      <td>{code}</td>
      <td><Button onClick={() => props.onClick(props.currency)}>{props.buttonText}</Button></td>
    </tr>
  )
};

export default Currency;
