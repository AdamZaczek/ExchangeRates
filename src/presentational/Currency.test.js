import React from 'react';
import renderer from 'react-test-renderer';
import Currency from './Currency';

test('renders correctly', () => {
	const currency = {currency: "dolar australijski", code: "AUD", bid: 2.8149, ask: 2.8717}
	const tree = renderer.create(
		<Currency currency={currency} />,
	).toJSON();
	expect(tree).toMatchSnapshot();
});