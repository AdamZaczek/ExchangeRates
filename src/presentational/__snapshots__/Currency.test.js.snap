// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<tr>
  <td>
    dolar australijski
  </td>
  <td>
    2.8717
  </td>
  <td>
    2.8149
  </td>
  <td>
    AUD
  </td>
  <td>
    <button
      onClick={[Function]}
    />
  </td>
</tr>
`;
